### PR TITLE
[datadog_synthetics_test] Support `unary` gRPC steps in multistep API tests

### PR DESCRIPTION
**Pull request by Drarig29**
This is a PR stack, here is how it's organized:
- https://github.com/DataDog/terraform-provider-datadog/pull/2445 (← **you are here**)
- https://github.com/DataDog/terraform-provider-datadog/pull/2457
- https://github.com/DataDog/terraform-provider-datadog/pull/2406
- https://github.com/DataDog/terraform-provider-datadog/pull/2459

---

This PR adds support for `unary` gRPC steps in multistep API tests.

What was already supported:
- `healthcheck` gRPC test
- `unary` gRPC test
- Multistep API test
  - `healthcheck` gRPC step

### CODE REVIEW

**Review by estherk15 — DISMISSED** [review]

**Review by etnbrd — APPROVED** [review]
LGTM 👍

**Review by HantingZhang2 — APPROVED** [review]

File: examples/resources/datadog_synthetics_test/resource.tf
Comment (Drarig29) [inline examples/resources/datadog_synthetics_test/resource.tf:1]: Aligned a bunch of things in those examples, mostly the names, tags, domains and ports

Comment (Drarig29) [reply examples/resources/datadog_synthetics_test/resource.tf:1]: cc. @etnbrd, as this PR had conflicts with one of your PRs (https://github.com/DataDog/terraform-provider-datadog/pull/2454)

File: datadog/resource_datadog_synthetics_test_.go [hunk 1]
```diff
@@ -799,6 +799,7 @@ func syntheticsTestOptionsList() *schema.Schema {
 
 func syntheticsTestAPIStep() *schema.Schema {
 	requestElemSchema := syntheticsTestRequest()
+	// In test `options_list` generally, but in `api_step.request_definition` for API steps.
```
Comment (etnbrd) [inline datadog/resource_datadog_synthetics_test_.go:802]:
```suggestion
	// In test `options_list` for single API tests, but in `api_step.request_definition` for API steps.
```

**Review by etnbrd — APPROVED** [review]
small nit, but LGTM

**Review by nkzou — APPROVED** [review]

**Pull request merged**